### DISCUSSION TITLE
Does rich work with runserver_plus?

### DISCUSSION THREAD

**Discussion by autoferrit** [Textualize/rich#1255] [Q&A]
In the `django_extensions` library there is a command, `runserver_plus` that when used, seems to ignore my `LOGGING` settings. Running `runserver` normally it works. This is what I have in my settings.
```python
LOGGING = {
    "version": 1,
    "disable_existing_loggers": False,
    "formatters": {"rich": {"datefmt": "[%X]"}},
    "handlers": {
        "console": {
            "class": "rich.logging.RichHandler",
            "formatter": "rich",
            "level": "INFO",
            # "rich_traceback": True,
        },
        "mail_admins": {
            "level": "ERROR",
            "filters": ["require_debug_false"],
            "class": "django.utils.log.AdminEmailHandler",
        },
    },
    "loggers": {
        "django": {
            "handlers": ["console"],
            "level": "INFO",
            "propagate": False,
        },
        "werkzeug": {
            "handlers": ["console"],
            "level": "DEBUG",
            "propagate": True,
        },
        "django.admin": {
            "handlers": ["mail_admins"],
            "level": "ERROR",
            "propagate": True,
        },
    },
}
```
Iv'e looked in the docs and can't find a way to make it work.

**Comment by willmcgugan** [comment]
Probably not a Rich issue. Logging config is fiddly. Try setting the console handler level to "DEBUG"

**Comment by autoferrit** [comment]
Turns out when you run this in the terminal window in pycharm it doesn't work very well. running it in the actual terminal it works fine.

**Reply by willmcgugan** [reply]
Ah. See the note here about PyCharm

https://rich.readthedocs.io/en/stable/introduction.html

**Reply by autoferrit** [reply]
Hmm. So I see this note
> PyCharm users will need to enable “emulate terminal” in output console option in run/debug configuration to see styled output.
But my run config does not have that option
![image](https://user-images.githubusercontent.com/89895/120884684-64fe7d00-c599-11eb-93a8-27e5f4a53c49.png)

**Reply by willmcgugan** [reply]
I’m only going on second hand information here. I’m not a PyCharm user. When this comes up people seem happy with the advice. Maybe they moved the option?

**Reply by autoferrit** [reply]
Yea thats what I thought but I couldn't find it. I realized, that when setting up a django specific run configuration, that option is not available. But if you do a generic python run configuration, that option is present. So My suggestion would be to use an updated wording.

> PyCharm users will need to enable “emulate terminal” in output console option in run/debug configuration to see styled output. You must use a `Python` run configuration as this option is not available in the `Django Server` configuration. Point the `Script Path` option to the `manage.py` and set parameters to `runserver` or `runserver_plus` if using `django_extensions`
